agent-lib/src/main/java/org/terracotta/angela/agent/client/RemoteClientManager.java

```diff
@@ -157,12 +160,26 @@ private String buildClasspath() throws IOException {
     }
 
     String agentClassName = Agent.class.getName().replace('.', '/');
-    String agentClassPath = Agent.class.getResource("/" + agentClassName + ".class").getPath();
-
-    if (agentClassPath.startsWith("file:")) {
-      sb.append(agentClassPath, "file:".length(), agentClassPath.lastIndexOf('!'));
+    URL agentUrl = Agent.class.getResource("/" + agentClassName + ".class");
```

Comment:
The original code, in a Windows environment, improperly constructs the path to the jar containing `Agent`.

Reply:
@aurbroszniowski : weird: we do have test on Windows though ? This issue should have been caught since a while: this code is there since beginning and was just moved when I refactored Angela's com layer.

Reply:
IIRC, the library for which the file path is misformed on Windows also appears elsewhere in the assembled classpath.  Not real sure why the file reference was added again but it's possible that, under some conditions, the `Agent` library is omitted from the classpath assembled above.